File: BE/MovieManager.Api/Models/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace MovieManager.Api.Models;

public class Movie
{
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Genre { get; set; }

    [Range(1, 5)]
    public int? Rating { get; set; }

    [Url]
    [MaxLength(2000)]
    public string? PosterUrl { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }
}

File: FE/src/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  genre?: string | null;
  rating?: number | null;
  posterUrl?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface MoviePayload {
  title: string;
  genre?: string | null;
  rating?: number | null;
  posterUrl?: string | null;
}

export interface MovieQuery {
  search?: string;
  genre?: string;
  sortBy?: 'title' | 'rating' | 'created';
  sortDirection?: 'asc' | 'desc';
}
